include/libtorrent/aux_/session_impl.hpp

```diff
@@ -1050,21 +1102,43 @@ namespace libtorrent
 			int m_outstanding_router_lookups = 0;
 #endif
 
-			void send_udp_packet_hostname(char const* hostname
+			void send_udp_packet_hostname(std::weak_ptr<utp_socket_interface> sock
+				, char const* hostname
 				, int port
 				, span<char const> p
 				, error_code& ec
 				, int flags);
 
-			void send_udp_packet(bool ssl
+			void send_udp_packet_hostname_listen(aux::session_listen_socket* sock
```

Comment:
I understand that without bind-to-device the interface may be selected based on the routing table, but I thought the source IP was still controlled by the socket's bound address. In the general case the source IP is all we care about here.